Lombiq.Hosting.Tenants.Maintenance/AdminMenu.cs

```diff
@@ -0,0 +1,33 @@
+using Lombiq.HelpfulLibraries.OrchardCore.Navigation;
+using Lombiq.Hosting.Tenants.Maintenance.Controllers;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Navigation;
+
+namespace Lombiq.Hosting.Tenants.Maintenance;
+
+public sealed class AdminMenu : AdminMenuNavigationProviderBase
+{
+    private readonly ShellSettings _shellSettings;
+
+    public AdminMenu(IHttpContextAccessor hca, IStringLocalizer<AdminMenu> stringLocalizer, ShellSettings shellSettings)
+        : base(hca, stringLocalizer) => _shellSettings = shellSettings;
+
+    protected override void Build(NavigationBuilder builder)
+    {
+        // Don't add the menu item on non-default tenants.
+        if (!_shellSettings.IsDefaultShell())
```

Comment:
Instead of hiding the menu item, this feature should only be allowed on the default tenant. Currently if you try to navigate to `/Admin/StaggeredMaintenance/Index` on an other tenant, you'll get an exception.

Reply:
Yeah I messed up, I added the feature to be enabled from Program.cs. I thought that DefaultTenantOnly in manifest actually does what I want.

Reply:
Ahh, this is a misery. Do we have experience in enabling a feature only for default tenant, but not from a recipe? Because even if I check if the tenant is the default tenant in the startup so the services are not added, the controller will be activated anyway. 

If I delete the activation of this feature from program.cs, then you need to enable the feature manually whenever you do a staging reset, until it is not in prod. Once it is in prod you have to manually enable the feature.

How should we do this?

Reply:
I think the easiest would be to enable the feature manually once on the default tenant.

Reply:
Actually,

> I thought that DefaultTenantOnly in manifest actually does what I want.

should work indeed. Did the controller belong to that feature only? Because it might be that it mostly did what you wanted, but not everything was covered.

Reply:
Yes, the controller had the Feature attribute as it has it now.

Reply:
Hmm. Does it work in vanilla Orchard? Or is it even used there this way (on a Controller)? I have a hunch that since ASP.NET Core has a different DI mechanism (compared to O1), it doesn't work this way, but it could be a bug too.

Reply:
I don't think there is a problem with the feature attribute on the controller. When the feature is not enabled you get 404 like you should. When the feature is enabled, you get the page.

If the feature is enabled through Program.cs (I think that doesn't respect the DefaultTenantOnly property in the manifest), the feature gets enabled on tenants also. But in the Startup of this feature I only let default tenant to add services and other stuff, but for the controller I can't do the same.

So either I check in the controller on each action if it's the default tenant and return 404 if it isn't, then I put back the feature enabling in the Program.cs so it gets enabled on all tenant.
Or we have to enable it manually once in prod.

Reply:
Ooooh, OK! Manually enabling it for the first time is no problem (make sure that it's in the setup recipe too).